encoder/irc_bit_allocation.c

```diff
@@ -521,11 +526,13 @@ WORD32 irc_ba_get_cur_frm_est_texture_bits(bit_allocation_t *ps_bit_allocation,
         number_t_to_word32(vq_est_texture_bits_for_frm,
                            &i4_est_texture_bits_for_frm);
 
-        i4_est_texture_bits_for_frm =
-                        (I_PIC == e_pic_type) ?
-                                        (i4_est_texture_bits_for_frm
-                                                        * I_TO_P_BIT_RATIO) :
-                                        i4_est_texture_bits_for_frm;
+        if(I_PIC == e_pic_type)
+        {
+            WORD64 i4_i_pic_bits =
```

Comment:
Ashwin,
We would need to investigate a bit more on why i4_i_pic_bits going beyond INT32_MAX.
what were the values of bitrate, framerate that lead to this high value and if something can be checked in svc encoder's configure level that would be ideal.
Also note, this doesn't show up in avc encoder, so maybe svc encoder is indeed missing some checks during initial set up?

Reply:
I will gather some data and see how this can be detected during init.

Reply:
I looked deeper and it seems like the issue was being caused by incorrect handling of constraints on the order of control commands, specifically set_dimensions and set_frame_rate. These are expected to occur in that order since the first one eventually triggers rc_init, which initialises frame_time struct used by a function eventually invoked by set_frame_rate.

Also, I noticed the CI fuzzer failures with this change. I am looking into them.